core/domain/imps_jobs_one_off.py

```diff
@@ -0,0 +1,179 @@
+                yield ('ENTITY_ID_ERROR', (task_key, entity_id_error))
+                return
+
+        # ENTITY_VERSION_ERROR check: unrecoverable.
+        if task.entity_version_start < version_end:
+            version_range = list(
+                python_utils.RANGE(task.entity_version_start, version_end))
+            try:
+                versioned_entities = entity_cls.get_multi_versions(
+                    task.entity_id, version_range)
+            except Exception:
+                entity_version_error = (
+                    '%s is not valid in the version range: [%s, %s)' % (
+                        entity_key, task.entity_version_start, version_end))
+                yield ('ENTITY_VERSION_ERROR', (task_key, entity_version_error))
+                return
+        else:
+            if task.entity_version_start == version_end:
+                entity_version_error = 'empty range: [%s, %s)' % (
+                    task.entity_version_start, version_end)
+            else:
+                entity_version_error = 'invalid range: [%s, %s)' % (
+                    task.entity_version_start, version_end)
+            yield ('ENTITY_VERSION_ERROR', (task_key, entity_version_error))
+            return
+
+        # Recoverable checks.
+        for version in version_range:
+            yield ('ENTITY_VERSION_COLLISION', (task_key, version))
+        for task_status_error in _get_task_status_errors(task):
+            yield ('TASK_STATUS_ERROR', (task_key, task_status_error))
+        for target_type_error in _get_target_type_errors(task):
+            yield ('TARGET_TYPE_ERROR', (task_key, target_type_error))
+        for target_id_error in _get_target_id_errors(task, versioned_entities):
+            yield ('TARGET_ID_ERROR', (task_key, target_id_error))
+
+    @staticmethod
+    def reduce(group, task_key_value_pairs):
+        """Checks that the given entity contains the expected relationships."""
+        task_key_values = collections.defaultdict(list)
+        for task_key_value_pair in task_key_value_pairs:
+            task_key, value = ast.literal_eval(task_key_value_pair)
+            task_key_values[task_key].append(value)
+
+        if group == 'ENTITY_VERSION_COLLISION':
+            version_coverage = collections.defaultdict(list)
+            for task_key, versions in task_key_values.items():
+                for version in versions:
+                    version_coverage[version].append(task_key)
+
+            task_key_values.clear()
+            for version, task_keys in version_coverage.items():
+                if len(task_keys) > 1:
+                    for task_key in task_keys:
+                        task_key_values[task_key].append(
+                            'collision at version %s' % version)
+
+        for task_key, values in task_key_values.items():
+            for value in values:
+                yield '%s at %s: %s' % (group, task_key, value)
```

Comment:
You can have the reduce function as it is and modify to check `cls.errors` keys and then check the relations.